### PR TITLE
Home: improve delay when switching story images

### PR DESCRIPTION
**Pull request by distantnative**
- Decreased the CSS transition time
- Removed the initial JS delay
- Shortened the final JS delay to match CSS transition time

### CODE REVIEW

File: assets/js/layouts/playground.js [hunk 1]
```diff
@@ -11,34 +9,37 @@ export class Playground {
...
+			.classList.add("loading");
+
+		// replace the playground
+		this.$el.innerHTML = doc.querySelector(".playground").innerHTML;
+
+		// fade in the image on load
+		this.$el
+			.querySelector(".playground-header-figure img")
+			.addEventListener("load", async function () {
+				await new Promise((resolve) => setTimeout(resolve, 100));
```
Comment (lukasbestle) [inline assets/js/layouts/playground.js]: TBH I don't fully understand what this delay is even trying to do, also in the old version. We had one delay that waits until the old image is faded out and one delay that waits for another 200 (now 100) ms before fading the new image in. But the actual fading-in happens *after* that second delay. So why do we need to wait here?

What I think we need is logic like "perform the fetch immediately and if less than 100 ms (= fading-out time of the old image) has passed by the time we are ready, wait until the fade-out was completed". So basically a `Promise.all()` of the fetch request and a 100 ms delay. Once both are finished, the new content can be injected and faded in.

File: assets/css/layouts/home.css [hunk 2]
```diff
@@ -95,7 +95,7 @@
 	transition: all 0.2s;
 }
 .playground-header-figure-wrapper img {
-	transition: all 0.3s;
+	transition: all 0.1s;
 }
 .playground-header-figure-wrapper.loading img {
 	opacity: 0;
```
Comment (lukasbestle) [inline assets/css/layouts/home.css:101]: Something is wrong about the transition. If I simulate a slow connection (200 ms extra wait time for the `fetch` request), I get flickering like this in Safari:

https://github.com/getkirby/getkirby.com/assets/1595007/82687b9d-70b9-4a29-afc4-ef0ce7ab8174

Other browsers don't seem to have that issue, but in Safari it happens all the time. Do you also see that?

**Comment by lukasbestle** [comment]
About preloading: I don't think it makes sense to preload all playground stories when visiting the homepage, but we could make it a bit faster by preloading the main story image immediately in the `switchTo` method when a new story is loaded.

Could work like this:

1. Add a custom `data-image` attribute to the [link in the menu](https://github.com/getkirby/getkirby.com/blob/de244bc76db0c2b1e08966d843b025ee003f8423/site/snippets/templates/home/playground/header.php#L38).
2. Before starting the `fetch` request for the HTML, do a `new Image().src = url;` to preload the image in the background.

File: assets/css/layouts/home.css [hunk 3]
```diff
@@ -95,7 +95,7 @@
 	transition: all 0.2s;
 }
 .playground-header-figure-wrapper img {
-	transition: all 0.3s;
+	transition: all 0.1s;
```
Comment (lukasbestle) [inline assets/css/layouts/home.css]: Testing this some more, I think I prefer 200 ms for the transition. It feels a bit more calm and also leaves us more time to load the HTML and main image.

File: assets/js/layouts/playground.js (hunk 1 above)
Comment (distantnative) [reply assets/js/layouts/playground.js]: You are right, and the `Promise.all()` sounds like a good direction. But I think it's not just fetch and the animation delay, but the image being loaded additionally.

Comment (distantnative) [reply assets/js/layouts/playground.js]: Ok, now fully understood it. Give me a few minutes.

**Comment by distantnative** [comment]
@lukasbestle I am also thinking if there would be a better way than fetch the whole home page when switching, just to extract the playground part. Maybe we could work with a content representation here that only return the playground part?

File: assets/css/layouts/home.css (hunk 2 above)
Comment (distantnative) [reply assets/css/layouts/home.css:101]: I have it in Firefox as well that sometime the fade-in animation is shown, but sometimes not at all.

Comment (lukasbestle) [reply assets/css/layouts/home.css:101]: Splitting the image loading and the fade-in to two separate ticks seems to do the trick.

**Comment by lukasbestle** [comment]
Made some additional enhancements to reduce flickering. I think we could merge it like this already, check the results in production and optimize further if needed. I feel like reducing the HTML payload would bring us a tiny improvement compared to the added complexity as it's just text and normal pages load quickly.

**Comment by distantnative** [comment]
I still experience sometimes the flicker. But it's the same on the old version. So still purely improvement. I agree on merging already and tweaking further later.

**Pull request merged**